clang/lib/Sema/SemaDecl.cpp

```diff
@@ -18377,6 +18377,15 @@ Decl *Sema::getObjCDeclContext() const {
   return (dyn_cast_or_null<ObjCContainerDecl>(CurContext));
 }
 
+Sema::DeviceDiagnosticReason Sema::getEmissionReason(const FunctionDecl *FD) {
+  if (FD->hasAttr<SYCLSimdAttr>())
+    return Sema::DeviceDiagnosticReason::Esimd;
+  else if (FD->hasAttr<SYCLDeviceAttr>() || FD->hasAttr<SYCLKernelAttr>())
+    return Sema::DeviceDiagnosticReason::Sycl;
+  // TODO: Figure out the logic for OMP and CUDA.
```

Comment:
Are you planning to implement this as part of this patch? If not, I think this should become a FIXME.

Reply:
Done.